src/Be.Vlaanderen.Basisregisters.CommandHandling/CommandHandlerResolver.cs

```diff
@@ -2,33 +2,30 @@ namespace Be.Vlaanderen.Basisregisters.CommandHandling
 {
     using System;
     using System.Collections.Generic;
+    using System.Linq;
 
     public class CommandHandlerResolver : ICommandHandlerResolver
     {
-        private readonly HashSet<Type> _knownCommandTypes = new HashSet<Type>();
-        private readonly Dictionary<Type, object> _handlers = new Dictionary<Type, object>();
+        private readonly Dictionary<Type, List<object>> _handlers = new Dictionary<Type, List<object>>();
 
         public CommandHandlerResolver(params CommandHandlerModule[] commandHandlerModules)
         {
             foreach(var module in commandHandlerModules)
             {
                 foreach(var handlerRegistration in module.HandlerRegistrations)
                 {
-                    if (!_knownCommandTypes.Add(handlerRegistration.CommandType))
-                        throw new InvalidOperationException(
-                            "Attempt to register multiple handlers for command type {0}".FormatWith(handlerRegistration.CommandType));
+                    if (!_handlers.ContainsKey(handlerRegistration.RegistrationType))
+                        _handlers[handlerRegistration.RegistrationType] = new List<object>();
```

Comment:
This allows to register duplicates of the same handler, not just registering multile different handlers for the same command type.

Shouldn't there be a guard to check if that specific handler is already registered?

Reply:
If you register a duplicate, wouldn't the `if ContainsKey` prevent it from being registered twice?